// nue-ui@0.7.0-bp.4/dist/es/popup-wrapper-DVF5gLbi.js
import { g as e, w as t } from "./utils-D3GAsA1N.js";
import { defineComponent as n, ref as p, onMounted as r, onBeforeUnmount as d, openBlock as i, createElementBlock as a } from "vue";
const u = { isExist: function() {
  return !!document.getElementById(this.id);
} }, o = /* @__PURE__ */ new Map();
function s() {
  if (u.element) return u.element;
  const t2 = document.createElement("div");
  return t2.classList.add("nue-popup-wrapper"), t2.id = e(), u.element = t2, document.body.appendChild(t2), t2;
}
function l(e2) {
  let t2;
  e2 && (t2 = o.get(e2)), t2 = !t2 && u.element ? u.element : s();
  const n2 = () => {
    const e3 = t2 == null ? void 0 : t2.childNodes.length;
    t2.dataset.actived = e3 ? "true" : "false";
  };
  return { appendChild: (e3) => {
    t2 == null ? void 0 : t2.appendChild(e3), n2();
  }, removeChild: (e3) => {
    t2 == null ? void 0 : t2.removeChild(e3), n2();
  } };
}
function m(e2) {
  return l(e2);
}
const c = ["id"], v = t(n({ name: "NuePopupWrapper", __name: "popup-wrapper", props: { id: {} }, setup(t2) {
  const n2 = p();
  return r(() => {
    var t3;
    n2.value && ((t3 = n2.value).id || (t3.id = e()), o.set(t3.id, t3));
  }), d(() => {
    var e2;
    n2.value && (e2 = n2.value, o.has(e2.id) ? o.delete(e2.id) : u.element === e2 && (u.element = void 0));
  }), (e2, t3) => (i(), a("div", { id: e2.id, ref_key: "popupWrapperRef", ref: n2, class: "nue-popup-wrapper" }, null, 8, c));
} }));
export {
  v as N,
  s as c,
  m as g,
  l as u
};
