// nue-ui@0.7.0-bp.4/dist/es/select-KdhfWKua.js
import { p as e, g as t, w as a } from "./utils-D3GAsA1N.js";
import { defineComponent as l, ref as o, computed as i, watch as s, onMounted as r, provide as p, openBlock as n, createBlock as d, unref as u, normalizeClass as c, withCtx as m, createVNode as v, withModifiers as f, createElementBlock as j, Fragment as b, createTextVNode as k, toDisplayString as g, createCommentVNode as h, renderSlot as x, inject as y } from "vue";
import "./avatar-yzliDfrH.js";
import "./badge-_zvsRtW_.js";
import { N as _ } from "./button-y8Aj-aBT.js";
import "./checkbox-B1odp8Rd.js";
import "./collapse-C_6fYhZ-.js";
import { N as z } from "./icon-DKepx2gf.js";
import { N } from "./text-TjewgIQ-.js";
import "./popup-wrapper-DVF5gLbi.js";
import "./confirm-YE1II1Pa.js";
import "./container-CYKikxGw.js";
import "./dialog-DDQvp2KT.js";
import "./div-Ds7FlWtC.js";
import "./drawer-C_6qrEkP.js";
import { N as S } from "./dropdown-CYLJA7-z.js";
import "./empty-9T0sEZ5j.js";
import "./infinite-scroll-DiID_Y5i.js";
import "./input-BhldYW3K.js";
import "./link-V-rQKoXz.js";
import "./progress-CLbjrDiX.js";
import "./prompt-CUXwrUXd.js";
import "./switch-BMjTISmX.js";
import "./tooltip-CPFII4M3.js";
import "./marquee-3hYdShqd.js";
import "./message-ChzyHR3L.js";
const w = l({ name: "NueSelect", __name: "select", props: { modelValue: {}, placeholder: { default: "Select" }, size: {}, disabled: { type: Boolean }, hideOnSelect: { type: Boolean, default: true }, clearable: { type: Boolean, default: false }, theme: {} }, emits: ["update:modelValue", "change"], setup(t2, { emit: a2 }) {
  const l2 = a2, y2 = t2, w2 = o([]), C2 = o(), V2 = i(() => {
    const { theme: t3 } = y2, a3 = "nue-select";
    let l3 = [a3];
    return t3 && (l3 = l3.concat(e(t3, a3))), l3;
  }), O2 = i(() => {
    var _a;
    return C2.value && ((_a = C2.value) == null ? void 0 : _a.label);
  });
  function B2(e2) {
    I(e2);
  }
  function I(e2, t3 = false) {
    let a3;
    if (t3) {
      if (C2.value && e2 === C2.value.value) return;
      for (const t4 of w2.value) if (e2 === t4.value) {
        a3 = t4;
        break;
      }
    } else {
      if (C2.value && e2 === C2.value.executeId) return;
      for (const t4 of w2.value) if (e2 === t4.executeId) {
        a3 = t4;
        break;
      }
      l2("update:modelValue", a3 == null ? void 0 : a3.value), l2("change", a3 == null ? void 0 : a3.value);
    }
    C2.value = a3 || void 0;
  }
  function q() {
    C2.value = void 0, l2("update:modelValue", void 0), l2("change", void 0);
  }
  return s(() => y2.modelValue, (e2) => {
    I(e2, true);
  }), r(() => {
    I(y2.modelValue, true);
  }), p("SelectContext", { optionRegister: function(e2) {
    w2.value.push(e2);
  }, selectedOption: C2 }), (e2, t3) => (n(), d(u(S), { class: c(V2.value), "hide-on-clicked": e2.hideOnSelect, "keep-alive": !!e2.$slots.default, size: e2.size, onExecute: B2 }, { default: m(({ clickTrigger: t4 }) => [v(u(_), { disabled: e2.disabled, size: e2.size, style: { gap: "16px" }, onClick: f(t4, ["stop"]) }, { default: m(() => [O2.value ? (n(), j(b, { key: 0 }, [k(g(O2.value), 1)], 64)) : (n(), d(u(N), { key: 1, color: "gray", style: { "font-size": "inherit !important" } }, { default: m(() => [k(g(e2.placeholder), 1)]), _: 1 }))]), append: m(() => [v(u(z), { name: "select" }), e2.clearable && C2.value ? (n(), d(u(z), { key: 0, name: "clear", onClick: f(q, ["stop"]) })) : h("", true)]), _: 2 }, 1032, ["disabled", "size", "onClick"])]), dropdown: m(() => [x(e2.$slots, "default")]), _: 3 }, 8, ["class", "hide-on-clicked", "keep-alive", "size"]));
} }), C = ["data-disabled", "data-executeid", "data-selected"], V = l({ name: "NueSelectOption", __name: "option", props: { label: {}, value: {}, disabled: { type: Boolean } }, setup(e2) {
  const a2 = y("SelectContext"), l2 = e2, s2 = o(t(4)), r2 = i(() => {
    const e3 = a2.selectedOption.value;
    return (e3 == null ? void 0 : e3.label) === l2.label;
  });
  function p2(e3) {
    l2.disabled && (e3.preventDefault(), e3.stopPropagation());
  }
  return a2.optionRegister({ executeId: s2.value, ...l2 }), (e3, t2) => (n(), j("li", { "data-disabled": e3.disabled, "data-executeid": s2.value, "data-selected": r2.value, class: "nue-select__option", onClick: p2 }, [k(g(e3.label) + " ", 1), r2.value ? (n(), d(u(z), { key: 0, name: "completed" })) : h("", true)], 8, C));
} }), O = a(w), B = a(V);
export {
  O as N,
  B as a
};
