// nue-ui@0.7.0-bp.4/dist/es/utils-D3GAsA1N.js
import { f as t, a as n } from "./vendor-DtiFC3Pk.js";
function r(n2) {
  return (r2) => t(n2, (t2) => {
    r2.use(t2);
  });
}
function e(t2) {
  return t2.install = (n2) => {
    const r2 = t2.name;
    n2.component(r2, t2);
  }, t2;
}
function o(t2, r2) {
  let e2 = [];
  return e2 = n(t2) ? t2.trim().split(",") : Array.isArray(t2) ? t2 : Object.keys(t2).filter((n2) => t2[n2]), e2 = e2.map((t3) => `${r2}--${t3.trim()}`), [].concat(e2);
}
function u(t2) {
  return "" === t2 ? "auto" : t2;
}
function a(t2) {
  return "" === t2 ? "wrap" : t2;
}
function i(t2) {
  return ("string" == typeof t2 || t2 instanceof String) && "[object String]" === Object.prototype.toString.call(t2);
}
function c(t2) {
  return "number" == typeof t2 && "[object Number]" === Object.prototype.toString.call(t2);
}
function s(t2) {
  return Array.isArray(t2);
}
function l(t2) {
  return "function" == typeof t2 && "function" == typeof t2.call;
}
function f(t2, n2) {
  let r2;
  return function(...e2) {
    r2 && clearTimeout(r2), r2 = setTimeout(() => {
      t2 && t2.call(null, ...e2), clearTimeout(r2), r2 = 0;
    }, n2);
  };
}
function m(t2, n2) {
  let r2;
  return (...e2) => {
    r2 || (t2 && t2.call(null, ...e2), r2 = setTimeout(() => {
      clearTimeout(r2), r2 = 0;
    }, n2));
  };
}
function p(t2 = 6) {
  return Math.random().toString(36).slice(2, 2 + t2);
}
export {
  u as a,
  l as b,
  a as c,
  f as d,
  c as e,
  s as f,
  p as g,
  i,
  r as m,
  o as p,
  m as t,
  e as w
};
