// nue-ui@0.7.0-bp.4/dist/es/vendor-DtiFC3Pk.js
var t = "object" == typeof global && global && global.Object === Object && global, e = "object" == typeof self && self && self.Object === Object && self, r = t || e || Function("return this")(), n = r.Symbol, o = Object.prototype, c = o.hasOwnProperty, u = o.toString, i = n ? n.toStringTag : void 0, a = Object.prototype.toString, f = n ? n.toStringTag : void 0;
function b(t2) {
  return null == t2 ? void 0 === t2 ? "[object Undefined]" : "[object Null]" : f && f in Object(t2) ? function(t3) {
    var e2 = c.call(t3, i), r2 = t3[i];
    try {
      t3[i] = void 0;
      var n2 = true;
    } catch (t4) {
    }
    var o2 = u.call(t3);
    return n2 && (e2 ? t3[i] = r2 : delete t3[i]), o2;
  }(t2) : function(t3) {
    return a.call(t3);
  }(t2);
}
function l(t2) {
  return null != t2 && "object" == typeof t2;
}
var p = Array.isArray;
function j(t2) {
  return t2;
}
function y(t2) {
  if (!function(t3) {
    var e3 = typeof t3;
    return null != t3 && ("object" == e3 || "function" == e3);
  }(t2)) return false;
  var e2 = b(t2);
  return "[object Function]" == e2 || "[object GeneratorFunction]" == e2 || "[object AsyncFunction]" == e2 || "[object Proxy]" == e2;
}
function s(t2, e2) {
  for (var r2 = -1, n2 = null == t2 ? 0 : t2.length; ++r2 < n2 && false !== e2(t2[r2], r2, t2); ) ;
  return t2;
}
var v = /^(?:0|[1-9]\d*)$/;
function g(t2, e2) {
  var r2 = typeof t2;
  return !!(e2 = null == e2 ? 9007199254740991 : e2) && ("number" == r2 || "symbol" != r2 && v.test(t2)) && t2 > -1 && t2 % 1 == 0 && t2 < e2;
}
function d(t2) {
  return "number" == typeof t2 && t2 > -1 && t2 % 1 == 0 && t2 <= 9007199254740991;
}
function O(t2) {
  return null != t2 && d(t2.length) && !y(t2);
}
var h = Object.prototype;
function A(t2) {
  return l(t2) && "[object Arguments]" == b(t2);
}
var m = Object.prototype, x = m.hasOwnProperty, S = m.propertyIsEnumerable, F = A(/* @__PURE__ */ function() {
  return arguments;
}()) ? A : function(t2) {
  return l(t2) && x.call(t2, "callee") && !S.call(t2, "callee");
}, T = "object" == typeof exports && exports && !exports.nodeType && exports, w = T && "object" == typeof module && module && !module.nodeType && module, P = w && w.exports === T ? r.Buffer : void 0, U = (P ? P.isBuffer : void 0) || function() {
  return false;
}, B = {};
B["[object Float32Array]"] = B["[object Float64Array]"] = B["[object Int8Array]"] = B["[object Int16Array]"] = B["[object Int32Array]"] = B["[object Uint8Array]"] = B["[object Uint8ClampedArray]"] = B["[object Uint16Array]"] = B["[object Uint32Array]"] = true, B["[object Arguments]"] = B["[object Array]"] = B["[object ArrayBuffer]"] = B["[object Boolean]"] = B["[object DataView]"] = B["[object Date]"] = B["[object Error]"] = B["[object Function]"] = B["[object Map]"] = B["[object Number]"] = B["[object Object]"] = B["[object RegExp]"] = B["[object Set]"] = B["[object String]"] = B["[object WeakMap]"] = false;
var I, k = "object" == typeof exports && exports && !exports.nodeType && exports, E = k && "object" == typeof module && module && !module.nodeType && module, q = E && E.exports === k && t.process, D = function() {
  try {
    return E && E.require && E.require("util").types || q && q.binding && q.binding("util");
  } catch (t2) {
  }
}(), M = D && D.isTypedArray, N = M ? (I = M, function(t2) {
  return I(t2);
}) : function(t2) {
  return l(t2) && d(t2.length) && !!B[b(t2)];
}, C = Object.prototype.hasOwnProperty, G = /* @__PURE__ */ function(t2, e2) {
  return function(r2) {
    return t2(e2(r2));
  };
}(Object.keys, Object), L = Object.prototype.hasOwnProperty;
function R(t2) {
  return O(t2) ? function(t3) {
    var e2 = p(t3), r2 = !e2 && F(t3), n2 = !e2 && !r2 && U(t3), o2 = !e2 && !r2 && !n2 && N(t3), c2 = e2 || r2 || n2 || o2, u2 = c2 ? function(t4, e3) {
      for (var r3 = -1, n3 = Array(t4); ++r3 < t4; ) n3[r3] = e3(r3);
      return n3;
    }(t3.length, String) : [], i2 = u2.length;
    for (var a2 in t3) !C.call(t3, a2) || c2 && ("length" == a2 || n2 && ("offset" == a2 || "parent" == a2) || o2 && ("buffer" == a2 || "byteLength" == a2 || "byteOffset" == a2) || g(a2, i2)) || u2.push(a2);
    return u2;
  }(t2) : function(t3) {
    if (r2 = (e2 = t3) && e2.constructor, e2 !== ("function" == typeof r2 && r2.prototype || h)) return G(t3);
    var e2, r2, n2 = [];
    for (var o2 in Object(t3)) L.call(t3, o2) && "constructor" != o2 && n2.push(o2);
    return n2;
  }(t2);
}
var V, W = (V = function(t2, e2) {
  return t2 && function(t3, e3, r2) {
    for (var n2 = -1, o2 = Object(t3), c2 = r2(t3), u2 = c2.length; u2--; ) {
      var i2 = c2[++n2];
      if (false === e3(o2[i2], i2, o2)) break;
    }
    return t3;
  }(t2, e2, R);
}, function(t2, e2) {
  if (null == t2) return t2;
  if (!O(t2)) return V(t2, e2);
  for (var r2 = t2.length, n2 = -1, o2 = Object(t2); ++n2 < r2 && false !== e2(o2[n2], n2, o2); ) ;
  return t2;
});
function $(t2, e2) {
  var r2;
  return (p(t2) ? s : W)(t2, "function" == typeof (r2 = e2) ? r2 : j);
}
function z(t2) {
  return "string" == typeof t2 || !p(t2) && l(t2) && "[object String]" == b(t2);
}
export {
  z as a,
  $ as f,
  y as i
};
